.github/workflows/cpp.yml

```diff
@@ -222,14 +224,20 @@ jobs:
         with:
           fetch-depth: 0
           submodules: recursive
+      - name: Enable ODBC build
+        run: |
+          echo "ARROW_FLIGHT_SQL=ON" >> $GITHUB_ENV
+          echo "ARROW_FLIGHT_SQL_ODBC=ON" >> $GITHUB_ENV
```

Comment:
I think this code block can be removed now, since these variables are already set under `env`